components/emoji_picker/utils/index.ts

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import isEmpty from 'lodash/isEmpty';
+
+import {isSystemEmoji} from 'mattermost-redux/utils/emoji_utils';
+import {Emoji, EmojiCategory, SystemEmoji} from 'mattermost-redux/types/emojis';
+
+import {
+    Categories,
+    Category,
+    CategoryOrEmojiRow,
+    CategoryHeaderRow,
+    EmojiRow,
+} from 'components/emoji_picker/types';
+
+import {EmojiIndicesByCategory, Emojis as EmojisJson, EmojiIndicesByUnicode} from 'utils/emoji.jsx';
+import {getSkin} from 'utils/emoticons';
+import {compareEmojis} from 'utils/emoji_utils';
+import EmojiMap from 'utils/emoji_map';
+
+import {
+    EMOJI_PER_ROW,
+    CATEGORY_HEADER_ROW,
+    EMOJIS_ROW,
+    SEARCH_RESULTS,
+    RECENT,
+    RECENT_EMOJI_CATEGORY,
+    CATEGORIES,
+} from 'components/emoji_picker/constants';
+
+export function isCategoryHeaderRow(row: CategoryOrEmojiRow): row is CategoryHeaderRow {
+    return row.type === CATEGORY_HEADER_ROW;
+}
+
+function sortEmojis(
```

Comment:
out of curiosity, what does this function actually do?

Reply:
simplified this whole function now its more readable.